pyscf/adc/uadc.py

```diff
@@ -607,127 +619,66 @@ def compute_energy(myadc, t1, t2, eris):
     nvir_a = myadc._nvir[0]
     nvir_b = myadc._nvir[1]
 
-    eris_ovov = eris.ovov
-    eris_OVOV = eris.OVOV
-    eris_ovOV = eris.ovOV
-
-    t2_1_a, t2_1_ab, t2_1_b  = t2[0]
+    eris_ovvo = eris.ovvo
+    eris_OVVO = eris.OVVO
+    eris_ovVO = eris.ovVO
 
-    ab_ind_a = np.tril_indices(nvir_a, k=-1)
-    ab_ind_b = np.tril_indices(nvir_b, k=-1)
+    t2_a  = t2[0][0][:].copy()
+    t2_ab  = t2[0][1][:].copy()
+    t2_b  = t2[0][2][:].copy()
 
-    #Compute MP2 correlation energy
-    e_mp2 = 0.25 * lib.einsum('ijab,iajb', t2_1_a, eris_ovov)
-    e_mp2 -= 0.25 * lib.einsum('ijab,ibja', t2_1_a, eris_ovov)
-    e_mp2 += lib.einsum('ijab,iajb', t2_1_ab, eris_ovOV)
-    e_mp2 += 0.25 * lib.einsum('ijab,iajb', t2_1_b, eris_OVOV)
-    e_mp2 -= 0.25 * lib.einsum('ijab,ibja', t2_1_b, eris_OVOV)
+    if (myadc.method == "adc(3)"):
+       
+       t2_a += t2[1][0][:].copy()
```

Comment:
Don't think copies are  necessary here since you're adding these arrays.

Reply:
Removed copies